src/src/rx_main.cpp

```diff
@@ -1727,6 +1727,17 @@ static void cycleRfMode(unsigned long now)
         Radio.RXnb();
         DBGLN("%u", ExpressLRS_currAirRate_Modparams->interval);
 
+#if defined(RADIO_LR1121)
+        // Skip Dual Band modes for hardware with only a single LR1121
+        while ((GPIO_PIN_NSS_2 == UNDEF_PIN && get_elrs_airRateConfig(scanIndex % RATE_MAX)->radio_type == RADIO_TYPE_LR1121_LORA_DUAL) ||
```

Comment:
Ill try and test this today BUT!

This only stops the Rx from cycling around and connecting to a blocked domain.  Users can still connect to a 2.4 only rx, then change modes to 900.  Now they will be blasting 100mW out the 900 pin which doesnt have a rf front end.

This was ok on my OG PR because accidentally changing to and X mode didnt cause harm.  But I think we now need a condition to block modes when a change is requested from the sync packet.

Reply:
Yes, I thought it was too easy and was sure there was something missing that I hadn't thought of.
I think the simplest fix would be to just drop the connection if the mode is not supported. Thoughts?

Reply:
Agree.  The RX should stay on the current RF mode and not change to a blocked mode.

Reply:
Extracted big check out to a common function and called that from both the TX and RX. Also fixed the RX not starting of the mode was not correct and skip the mode change in `SetRFLinkRate`.